src/device/pf_cmrdr.c

```diff
@@ -123,10 +124,56 @@ int pf_cmrdr_rm_read_ind (
    }
    else
    {
+      /* Logging */
+      if (
+         (p_read_request->index >= PF_IDX_SUB_DIAGNOSIS_CH) &&
+         (p_read_request->index <= PF_IDX_SUB_DIAG_MAINT_DEM_ALL))
+      {
+         index_range_text = "subslot";
+      }
+      else if (
+         (p_read_request->index >= PF_IDX_SLOT_DIAGNOSIS_CH) &&
+         (p_read_request->index <= PF_IDX_SLOT_DIAG_MAINT_DEM_ALL))
+      {
+         index_range_text = "slot";
+      }
+      else if (
+         (p_read_request->index >= PF_IDX_API_DIAGNOSIS_CH) &&
+         (p_read_request->index <= PF_IDX_API_DIAG_MAINT_DEM_ALL))
+      {
+         index_range_text = "api";
+      }
+      else if (
+         (p_read_request->index >= PF_IDX_AR_DIAGNOSIS_CH) &&
+         (p_read_request->index <= PF_IDX_AR_DIAG_MAINT_DEM_ALL))
+      {
+         index_range_text = "ar";
+      }
+      else if (p_read_request->index == PF_IDX_DEV_DIAGNOSIS_DMQS)
+      {
+         index_range_text = "device";
+      }
+      (void)index_range_text; /* Avoid not-used-variable warning if no log */
```

Comment:
osal_log.h now has a LOG_INFO_ENABLED macro that you could wrap the logging code in.

Reply:
Done